### PR TITLE
Remove common-labels

### LINKED ISSUE #3581: Improve UX of (or simply remove) "Common labels" from LogCLI
**Problem description**

I'm piping output from `logcli` into other Unix tools at the command-line (specifically [within a Python script](https://github.com/grafana/deployment_tools/blob/master/docker/bigquery-sync/loki/loki-tap.py#L24-L36)). Downstream (down-pipe?) I want to manipulate **all** labels that I have selected/renamed in the `logcli` query via `logfmt` and `label_format`. But unfortunately those labels I am renaming are not present in the line-by-line output on STDOUT, because they are common across all records! They're being omitted from STDOUT and moved into the "Common labels:" output on STDERR (where they aren't accessible to other tools that I'm piping STDOUT into).

**Some options for solution I'd like to see**
- Docs
   - Make the docs more explicit - highlight this functionality more prominently. It's easy to miss the "Common labels" output on STDERR - especially if looking at 30 records on screen, or (worse) when piping directly into `jq`
   - From @cyriltovena on [Slack](https://raintank-corp.slack.com/archives/C9T1FLN9K/p1617096921379300?thread_ts=1617093659.373100&cid=C9T1FLN9K): "`-o jsonl` doc is lying it also uses common labels"
- UX
   - Possibly remove the "Common labels" functionality completely. This may be a breaking change, but it seems the simplest solution to me at the moment. I think it was added by @tomwilkie in the very early days of this project here: [`687cb06` (#7)](https://github.com/grafana/loki/pull/7/commits/687cb0633532e7b326727cd3ec31b1e31735c8e9), and doesn't seem to have been added in response to any external or community requests.
   - Add a CLI flag to allow the "Common labels" functionality to be switched off by the user.
   - When renaming labels via `label_format` we could consider always adding these labels to the log lines themselves, rather than adding them to "Common labels". For me at least I was using `label_format` because I wanted to see those labels in STDOUT.

**Alternatives I've considered**
We (@jeschkies and I) briefly considered parsing STDERR as well as STDOUT, but we wanted to see what others thought about this issue first.

**Additional context**
Initial [Slack discussion](https://raintank-corp.slack.com/archives/C9T1FLN9K/p1617096453376700?thread_ts=1617093659.373100&cid=C9T1FLN9K) on Grafana internal

The command I've been running is this:
```
logcli -o jsonl query '{name="cortex-gw", namespace="prod"} |= "eventType=bi" | logfmt | label_format user_agent=userAgent,user_agent_group=userAgentGroup,ns=namespace' | jq .
```

And specifically the `ns=namespace` is coming out on STDERR (when I remove the `jq` downstream pipe):
Screenshot 2021-04-06 at 13 47 42

### PR DESCRIPTION
**Pull request by samjewell**
Fixes #3581

This common-labels functionality made the CLI tool harder to use when
piping the output into other programs. The common labels would be printed
on STDERR, while the rest of the labels and log-lines were emitted on
STDOUT.
I don't think we need this functionality any more, now that we have the
`--exclude-label` option.

**Checklist**
- [x] Documentation ~~added~~ removed and updated
- [x] Tests updated

### CODE REVIEW

File: pkg/logcli/query/query.go
Comment (sandeepsukhani) [reply pkg/logcli/query/query.go:262]: Looks like we already skip printing common labels in Quiet mode https://github.com/grafana/loki/pull/4522/files#diff-1a68a4d189f6dc4eda3cffe8ba6735d4480c95fedde0ba9dc167adba3c48cc44L269
Was it not working before?

File: docs/sources/getting-started/logcli.md
Comment (KMiller-Grafana) [inline docs/sources/getting-started/logcli.md:159]:
```suggestion
    results, such as the API URL and a set of excluded labels.
    This extra information can be suppressed with the `--quiet` flag.
```

File: docs/sources/getting-started/logcli.md
Comment (KMiller-Grafana) [inline docs/sources/getting-started/logcli.md:237]:
```suggestion
results, such as the API URL and a set of excluded labels.
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]
Small changes to improve the grammar.

File: pkg/logcli/query/query.go
Comment (samjewell) [reply pkg/logcli/query/query.go:262]: @sandeepsukhani yes I can suppress the common labels in Quiet-mode - this does still work. But that's not what I want here. I instead want to ensure that **all** labels are present in the output on STDOUT, so that I can pipe the output from logcli into other command-line programs. See the original PR description and the issue referenced. 

Basically I want logcli to adhere more closely to the Unix philosophy. I'll use other unix programs downstream in the pipe as I see fit to further edit and filter log lines - I don't want logcli to be "too clever" here.

**Review by jeschkies — APPROVED** [review]

**Comment by samjewell** [comment]
@KMiller-Grafana can I have another review please? Your "request changes" is blocking this from being merged, as far as I can tell.

**Review by KMiller-Grafana — APPROVED** [review]
Docs portion of this LGTM now.  Thanks for the changes.

**Comment by samjewell** [comment]
@cyriltovena or @sandeepsukhani I still need a maintainer approval. Would either of you be happy to approve, then merge this? 🙏

**Review by slim-bean — APPROVED** [review]
I think this makes sense, I think logcli is being used more frequently as a tool for feeding other systems and this change makes sense to me.

I know personally I rarely run a query without `--no-labels` when using logcli for searching logs because i really rarely care what stream the log lines came from so this would have no impact on me.

However, this is a big change in behavior and I think @samjewell we should put something in the upgrade guide to note the change and suggest `--no-labels` can be used to simplify the output, WDYT?

**Comment by kavirajk** [comment]
Closing this as no activities over year(s). Feel free to send new PR if anyone want's to revive the work

**Pull request closed without merging**
